File: id/src/main/java/com/simprints/id/data/db/remote/models/fb_IdEventUpdate.java
package com.simprints.id.data.db.remote.models;

import android.support.annotation.Nullable;

import com.google.firebase.database.ServerValue;
import com.simprints.id.data.db.remote.tools.Utils;

import java.util.Map;

@SuppressWarnings("WeakerAccess")
public class fb_IdEventUpdate {
    public String selectedId;
    public String projectKey;
    public String androidId;
    public Long date;
    public String sessionId;
    public Map<String, String> serverDate;

    public fb_IdEventUpdate() {
    }

    public fb_IdEventUpdate(@Nullable String apiKey, String selectedId, String androidId, String sessionId) {
        if (apiKey != null)
            this.projectKey = apiKey;

        this.selectedId = selectedId;
        this.androidId = androidId;
        this.date = Utils.now().getTime();
        this.sessionId = sessionId;
        this.serverDate = ServerValue.TIMESTAMP;
    }
}

File: id/src/main/java/com/simprints/id/data/db/remote/models/fb_IdEventUpdate.kt
package com.simprints.id.data.db.remote.models

import com.google.firebase.database.ServerValue
import com.simprints.id.data.db.remote.tools.Utils

class fb_IdEventUpdate(
    var selectedId: String,
    var projectKey: String,
    var androidId: String,
    var date: Long? = null,
    var sessionId: String,
    var serverDate: Map<String, String>) {

    constructor(projectKey: String?,
                selectedId: String,
                androidId: String,
                sessionId: String): this (
        projectKey = projectKey ?: "",
        selectedId = selectedId,
        androidId = androidId,
        date = Utils.now().time,
        sessionId = sessionId,
        serverDate = ServerValue.TIMESTAMP)
}
